voice/voice-ai/x/ReviewRequest/ReviewRequester.swift

```diff
@@ -0,0 +1,76 @@
+
+import Foundation
+import StoreKit
+
+class ReviewRequester {
+    static let shared = ReviewRequester()
+    
+    // Configuration
+    private var minimumSignificantEvents: Int = 0
+    private var daysBetweenPrompts: Int = 0
+    
+    // User default keys
+    private let reviewRequestCountKey = "reviewRequestCount"
+    private let lastReviewRequestDateKey = "lastReviewRequestDate"
+    private let significantEventsCountKey = "significantEventsCount"
+    
+    // Properties
+    private var reviewRequestCount: Int {
+        get { UserDefaults.standard.integer(forKey: reviewRequestCountKey) }
+        set { UserDefaults.standard.set(newValue, forKey: reviewRequestCountKey) }
+    }
+    
+    private var lastReviewRequestDate: Date? {
+        get { UserDefaults.standard.object(forKey: lastReviewRequestDateKey) as? Date }
+        set { UserDefaults.standard.set(newValue, forKey: lastReviewRequestDateKey) }
+    }
+    
+    internal var significantEventsCount: Int {
+        get { UserDefaults.standard.integer(forKey: significantEventsCountKey) }
+        set { UserDefaults.standard.set(newValue, forKey: significantEventsCountKey) }
+    }
+    
+    // Singleton initialization with configurable parameters
+    static func initialize(minimumSignificantEvents: Int, daysBetweenPrompts: Int) {
+        ReviewRequester.shared.minimumSignificantEvents = minimumSignificantEvents
+        ReviewRequester.shared.daysBetweenPrompts = daysBetweenPrompts
+    }
+    
+    func logSignificantEvent() {
+        print("[ReviewRequester][logSignificantEvent]")
+        significantEventsCount += 1
+        tryPromptForReview()
+    }
+    
+    internal func tryPromptForReview() {
+        guard shouldPromptForReview() else { return }
+        
+        if let windowScene = UIApplication.shared.connectedScenes.first as? UIWindowScene {
+            SKStoreReviewController.requestReview(in: windowScene)
+            reviewRequestCount += 1
+            lastReviewRequestDate = Date()
+            resetSignificantEventsCount() // Reset the count after prompting for review
+        }
+    }
+    
+    private func shouldPromptForReview() -> Bool {
```

Comment:
Where is it checking whether a review is already written?

Reply:
I think apple has not provided any API to check review is already written.